KsApi/models/Reward+RewardAddOnSelectionView.swift

```diff
@@ -0,0 +1,86 @@
+import Foundation
+
+public extension Reward {
+  /**
+   Create an add-on reward from a RewardAddOnSelectionViewEnvelope.Project.Reward
+
+    - parameter reward: The RewardAddOnSelectionViewEnvelope.Project.Reward data structure.
+    - parameter project: The associated Project model.
+    - parameter selectedAddOnQuantities: The selected quantity for this add-on.
+    - parameter dateFormatter: A DateFormatter configured with the format "yyyy-MM-DD".
+
+    - returns: A Reward.
+   */
+
+  static func addOnReward(
+    from reward: RewardAddOnSelectionViewEnvelope.Project.Reward,
+    project: Project,
+    selectedAddOnQuantities: [String: Int],
+    dateFormatter: DateFormatter
+  ) -> Reward? {
+    guard let rewardId = decompose(id: reward.id) else { return nil }
+
+    let estimatedDeliveryOn = reward.estimatedDeliveryOn
+      .flatMap(dateFormatter.date(from:))?.timeIntervalSince1970
+
+    let addOnData = AddOnData(
+      isAddOn: true,
+      selectedQuantity: selectedAddOnQuantities[reward.id] ?? 0
+    )
+
+    return Reward(
+      addOnData: addOnData,
+      backersCount: reward.backersCount,
+      convertedMinimum: reward.convertedAmount.amount,
+      description: reward.description,
+      endsAt: reward.endsAt,
+      estimatedDeliveryOn: estimatedDeliveryOn,
+      hasAddOns: false, // This value is only sent via the v1 API to indicate that a base reward has add-ons
```

Comment:
As a reminder that this will need to updated when the V1 API is updated

Reply:
This function is used to 'adapt' the GraphQL Reward data to our v1 `Reward` model which we use throughout the app, so this comment is more to indicate that this is v1 data that we shouldn't expect here. If it is returned from GraphQL then we can remove this.